packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -374,15 +456,58 @@ where
         let all_msgs = update_msgs
             .into_iter()
             .map(|m| Any::from_msg(&m))
-            .chain(timeout_msgs.into_iter().map(|m| Any::from_msg(&m)))
-            .chain(recv_msgs.into_iter().map(|m| Any::from_msg(&m)))
-            .chain(ack_msgs.into_iter().map(|m| Any::from_msg(&m)))
+            .chain(timeout_msgs.clone().into_iter().map(|m| Any::from_msg(&m)))
+            .chain(recv_msgs.clone().into_iter().map(|m| Any::from_msg(&m)))
+            .chain(ack_msgs.clone().into_iter().map(|m| Any::from_msg(&m)))
```

Comment:
In order to not clone these, you can either save the lengths, or do the following:
```suggestion
            .chain(timeout_msgs.iter().map(|m| Any::from_msg(m)))
            .chain(recv_msgs.iter().map(|m| Any::from_msg(m)))
            .chain(ack_msgs.iter().map(|m| Any::from_msg(m)))
```